classes/proxy_lock_factory.php

```diff
@@ -325,6 +335,12 @@ private function log_update_count($record) {
             $history->released = time();
             $history->duration += $record->duration;
 
+            if (strpos($record->resourcekey, 'adhoc') !== false) {
+                $adhocid = explode('_', $record->resourcekey);
+                $timequeued = $DB->get_record('task_adhoc', array('id' => $adhocid[1]), 'nextruntime');
```

Comment:
Do we need this lookup?

isn't it just:

$history->latency += $record->latency; ?